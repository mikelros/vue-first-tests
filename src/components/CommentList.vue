<template>
  <div class="list">
    <h1>Comments</h1>
    <ol class="comments" v-if="comments.length > 0">
      <Comment v-for="(comment, i) in comments"
               :comment="comment"
               :key="i"
      />
    </ol>
    <p v-else>There are no comments yet!</p>
  </div>
</template>

<script>
import Comment from './Comment.vue';

export default {
  name: 'CommentList',
  props: {
    comments: {
      type: Array
    }
  },
  components: {
    Comment
  }
}
</script>

<style scoped>
  .list {
    width: 400px;
    margin: auto;
  }

  .comments {
    text-align: left;
  }
</style>
