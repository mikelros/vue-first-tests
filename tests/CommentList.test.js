import { mount } from 'vue-test-utils';
import CommentList from './../src/components/CommentList.vue';

describe('Comment List', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(CommentList, {
      propsData: {
        comments: []
      }
    });
  });

  it('has expected structure when no comments given', () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it('has expected structure when some comments given', () => {
    cmp.vm.comments = ['this', 'is', 'a', 'test'];
    cmp.update();

    expect(cmp.element).toMatchSnapshot();
  });
});
