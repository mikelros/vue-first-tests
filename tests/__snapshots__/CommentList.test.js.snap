// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment List has expected structure when no comments given 1`] = `
<div
  class="list"
>
  <h1>
    Comments
  </h1>
   
  <p>
    There are no comments yet!
  </p>
</div>
`;

exports[`Comment List has expected structure when some comments given 1`] = `
<div
  class="list"
>
  <h1>
    Comments
  </h1>
   
  <ol
    class="comments"
  >
    <li>
      
  this

    </li>
    <li>
      
  is

    </li>
    <li>
      
  a

    </li>
    <li>
      
  test

    </li>
  </ol>
</div>
`;
